components/region_cache_memory_engine/src/background.rs

```diff
@@ -364,35 +396,44 @@ impl Runnable for BackgroundRunner {
                 }
             }
             BackgroundTask::LoadRange => {
+                if self.core.memory_controller.reached_soft_limit() {
+                    // We are running out of memory, so not to load new range.
+                    return;
```

Comment:
should on_snapshot_load_canceled be called as well?

Reply:
I adjusted this and each load will check separately now.